### DISCUSSION TITLE
Any experience with VS Performance Profiler for StereoKit debugging?

### DISCUSSION THREAD

**Discussion by austinbhale** [StereoKit/StereoKit#605] [Q&A]
Heyo, I made an [issue in the VS help forum](https://developercommunity.visualstudio.com/t/Performance-Profiler-Exception-for-debug/10311837?) but wanted to see if anyone has a workaround for profiling their UWP apps.

When I try to enable Diagnostic Tools and the Performance Profiler, it'll throw an exception but still deploy. Has anyone been able to successfully profile their HoloLens 2 (UWP) apps? Thanks so much!

**Comment by maluoi** [comment]
I'm afraid I don't have terribly much to add here myself! I do the vast majority of my profiling while running on PC, which while not exact, tends to be pretty representative in the majority of my own cases. I'm actually not sure if I even tried on HL2 or not? I know that's not really helpful for on-device-only sort of stuff though :(

I might see if I can repro this on a HoloLens emulator later, if I can manage to get that working again!

**Reply by austinbhale** [reply]
Profiling on the PC would be ideal. But that's right, it's pretty limiting not being able to profile hardware-specific encoding/decoding techniques.

Turning on/off Hyper-V was one of the things I tried to resolve this issue. Hadn't tried on the emulator itself though. Ime it's super slow :D but worth a shot!
